scripts/configuration.js

```diff
@@ -89,36 +56,38 @@ async function populateBuildBranchDropdown(definitionRepository, codeClient, set
  */
 async function onBuildDropdownChange(definitions, codeClient)
 {
+
     console.debug("Resetting the branch dropdown");
-    branchDropdown.innerHTML = defaultBranchDropdownOptions;
-    branchDropdown.value = "";
-    let buildDefinition = buildDropdown.value
+    branchDropdown.innerHTML = '<option value="all" selected>all</option>';
+    branchDropdown.value = "all";
+
 
+    let buildDefinition = buildDropdown.value
     let currentDefinition = definitions.find(d => Number(d.id) === Number(buildDefinition));
 
-    let definitionRepository = currentDefinition.repository.id;
+    console.debug(`Retrieving the branches for the repository ${currentDefinition.repository.id}`);
+    fillBranchesDropDown(currentDefinition.repository.id, codeClient);
```

Comment:
Adding awaiter on async function call

```suggestion
    await fillBranchesDropDown(currentDefinition.repository.id, codeClient);
```

Reply:
I just wanted to ask if here an await is requried, and yes you were quicker than me :-)